corehq/util/test_utils.py

```diff
@@ -60,6 +60,27 @@ def trap_extra_setup(*exceptions, **kw):
         raise
 
 
+def softer_assert(func=None):
```

Comment:
Could use [ContextDecorator](https://github.com/dimagi/commcare-hq/blob/0020c7a9c811ecc3f3dfcb2ca182493318774810/corehq/util/decorators.py#L10-L10) to make this simpler